src/nwb_conversion_tools/tools/hdmf.py

```diff
@@ -1,7 +1,77 @@
+"""Collection of modifications of HDMF functions that are to be tested/used on this repo until propagation upstream."""
 from typing import Tuple
 
 import numpy as np
-from hdmf.data_utils import GenericDataChunkIterator
+from hdmf.data_utils import GenericDataChunkIterator as HDMFGenericDataChunkIterator
+
+
+class GenericDataChunkIterator(HDMFGenericDataChunkIterator):
+    def _get_default_buffer_shape(self, buffer_gb: float = 1.0) -> Tuple[int]:
+        chunk_bytes = np.prod(self.chunk_shape) * self.dtype.itemsize
+        assert buffer_gb > 0, f"buffer_gb ({buffer_gb}) must be greater than zero!"
+        assert (
+            buffer_gb >= chunk_bytes / 1e9
+        ), f"buffer_gb ({buffer_gb}) must be greater than the chunk size ({chunk_bytes / 1e9})!"
+        assert all(
+            np.array(self.chunk_shape) > 0
+        ), f"Some dimensions of chunk_shape ({self.chunk_shape}) are less than zero!"
+
+        maxshape = np.array(self.maxshape)
+
+        # Early termination condition
+        if np.prod(maxshape) * self.dtype.itemsize / 1e9 < buffer_gb:
+            return tuple(self.maxshape)
+
+        buffer_bytes = chunk_bytes
+        axis_sizes_bytes = maxshape * self.dtype.itemsize
+        smallest_chunk_axis, second_smallest_chunk_axis, *_ = np.argsort(self.chunk_shape)
+        target_buffer_bytes = buffer_gb * 1e9
+
+        # If the smallest full axis does not fit within the buffer size, form a square along the two smallest axes
+        sub_square_buffer_shape = np.array(self.chunk_shape)
+        if min(axis_sizes_bytes) > target_buffer_bytes:
+            k1 = np.floor((target_buffer_bytes / chunk_bytes) ** 0.5)
+            for axis in [smallest_chunk_axis, second_smallest_chunk_axis]:
+                sub_square_buffer_shape[axis] = k1 * sub_square_buffer_shape[axis]
+            return tuple(sub_square_buffer_shape)
+
+        # Original one-shot estimation has good performance for certain shapes
+        k2 = np.floor(
+            (buffer_gb * 1e9 / (np.prod(self.chunk_shape) * self.dtype.itemsize)) ** (1 / len(self.chunk_shape))
```

Comment:
```suggestion
            (buffer_gb * 1e9 / chunk_bytes) ** (1 / len(self.chunk_shape))
```

I would write:
```python
chunks_in_buffer = np.floor(buffer_gb * 1e9 / chunk_bytes)
chunks_per_axis  = chunks_in_buffer ** (1 / self.chunk_shape.ndim)
```
Instead of just k. But I have seen that they have this k values in the `hdfm` code. Up to you.

Reply:
adjusted naming and added steps